framework/src/media/StreamHandler.cpp

```diff
@@ -0,0 +1,191 @@
+/* ****************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include "StreamHandler.h"
+
+#include <debug.h>
+
+#ifndef CONFIG_HANDLER_STREAM_BUFFER_SIZE
+#define CONFIG_HANDLER_STREAM_BUFFER_SIZE 4096
+#endif
+
+#ifndef CONFIG_HANDLER_STREAM_BUFFER_THRESHOLD
+#define CONFIG_HANDLER_STREAM_BUFFER_THRESHOLD 2048
+#endif
+
+namespace media {
+namespace stream {
+
+StreamHandler::StreamHandler() :
+	mWorker(0),
+	mWorkerName("StreamWorker"),
+	mWorkerStackSize(4096),
+	mIsWorkerAlive(false)
+{
+}
+
+bool StreamHandler::open()
```

Comment:
Let's establish definition of each api.
1. Kind of Open means that ready to use. so you should prepare everything here, of course thread should wait there, it can be started when start called. Same opinion for stop & close.
2. If we allow stop during process running and start again(pause -> resume in player), cleaning is needed based on comment above.
3. 'Flush' also public method, so player can flush for remaining buffer to playback & capture, how do you think about it?

Reply:
I think these are also changing the handler's operation.
I'll raise another PR one by one.